clang/docs/checkedc/Setup-and-Build.md

```diff
@@ -57,138 +62,223 @@ need to end with line feeds. Visual Studio preserves line endings for files, so
-build.  This is because LLVM is being built.   The debug build of LLVM is
-particularly slow because it bottlenecks on table generation. LLVM generates
-architecture-specific tables at build time that are used during code
-generation.  The default table generation algorithm is very slow in debug
-builds.  Subsequent builds during development will be much faster (minutes, not
-an hour).
-
-### On UNIX
-Change to your build directory and build `clang`:
-```
-  ninja clang
-```
-
-### On Windows
-
-For day-to-day development, we recommend building from Visual Studio.  This
-will improve your productivity significantly because it will give you all the
+## Checkout and Build Instructions for Checked C Compiler
+
+You will need to choose a drive that has at least 50 Gbytes free. You may need
+lots of space for the sources and the build. You can store the sources in any
+directory that you want. You should avoid spaces in parent directory names
+because this can confuse some tools.
+
+### Instructions for Linux:
+
+1. Choose any directory as your working directory. We will refer to to this
+directory as \<WORK_DIR\>.
+
+   ```
+   cd <WORK_DIR>
+   ```
+
+2. Clone the `checkedc-clang` repository:
+
+   ```
+   git clone https://github.com/Microsoft/checkedc-clang src
+   ```
+
+3. The Checked C language tests live in a folder within `llvm/project`. Change
+to the  `src/llvm/projects/checkedc-wrapper` directory and clone the Checked C
+repo:
+   ```
+   git clone https://github.com/Microsoft/checkedc
+   ```
+
+4. **\[OPTIONAL\]** Install `ccache` to speed up the compiler build on Linux and
+MacOS. [ ccache](https://ccache.samba.org) is a smart cache for GCC or Clang. It
+works as a shim, and uses the hash of source files and their included headers
+and build options to decide if an output needs recompiling, instead of file
+modification time (which Make uses). In some circumstances, this can cut
+second-build (i.e. `make` where some of the files are already built) time down
+from 5 minutes to 30 seconds. This still depends on how your header files and
+includes are organized. Moreover, there are ways to share and control the size
+of the cache directory, which is where `ccache` stores a copy of any object
+files it has compiled.
+
+5. LLVM and Clang use `CMake`, which is a meta-build system generator. It
+generates build systems for a specific platform. Create a build directory that
+is a sibling of your LLVM source tree, like \<WORK_DIR\>/build.
+
+   ```
+   cd <WORK_DIR>/build
+   ```
+
+6. Execute the following `cmake` command in the build directory:
+
+   ```
+   cmake -G Ninja -DLLVM_ENABLE_PROJECTS=clang   // Required to enable Clang build
+   -DCMAKE_INSTALL_PREFIX=<WORK_DIR>/install     // Directory where the compiler will be
+                                                 // installed when "ninja install" is executed. 
+   -DCMAKE_BUILD_TYPE=Release                    // Alternate value: Debug
```

Comment:
Other supported values for `CMAKE_BUILD_TYPE` are `RelWithDebInfo` and `MinSizeRel`